### PR TITLE
Release persistent devnet on release tags

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**

### CODE REVIEW

File: .github/workflows/persistent-devnet.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
         description: "Release tag, branch name or any revision."
         required: true
         type: string
+  push:
+    tags:
+      - release-v[0-9].[0-9]
```
Comment (vimukthi-git) [inline .github/workflows/persistent-devnet.yml]: We want the tags to be on patch versions, also removing "release" prefix

```suggestion
      - v[0-9].[0-9].[0-9]
```
Eg: v2.4.1

File: .github/workflows/release-artifacts.yml [hunk 2]
```diff
@@ -200,10 +200,10 @@ jobs:
...
             # Spin up a local devnet (Dali + Rococo)
-            nix run github:ComposableFi/composable/v${{ env.COMPOSABLE_VERSION }}#devnet-up
+            nix run github:ComposableFi/composable/v${{ env.COMPOSABLE_VERSION }}#devnet
 
             # Spin up a local XCVM devnet (Dali + Rococo + Juno)
-            nix run github:ComposableFi/composable/v${{ env.COMPOSABLE_VERSION }}#devnet-xcvm-up
+            nix run github:ComposableFi/composable/v${{ env.COMPOSABLE_VERSION }}#devnet-xcvm
 
             # Show all possible apps, shells and packages
             nix flake show github:ComposableFi/composable/v${{ env.COMPOSABLE_VERSION }} --allow-import-from-derivation
```
Comment (vimukthi-git) [inline .github/workflows/release-artifacts.yml:209]: We would also like the release artifacts for wasm to include the runtime spec_version, according to the pattern

vA.B.C -> picasso-ABC 
Eg:
v2.4.2 -> picasso-242

Comment (vimukthi-git) [reply .github/workflows/release-artifacts.yml:209]: We will make sure that this is the actual spec_version by hand for now

Comment (vimukthi-git) [reply .github/workflows/release-artifacts.yml:209]: But we can automate later

File: .github/workflows/persistent-devnet.yml [hunk 3]
```diff
@@ -7,6 +7,9 @@ on:
         description: "Release tag, branch name or any revision."
         required: true
         type: string
+  push:
```
[the remark this answers is not in the document]
Comment (hussein-aitlahcen) [reply .github/workflows/persistent-devnet.yml:10]: I'm unsure about this, the tag is generated from the release artifacts workflow. And the release artifacts workflow is ran when a new release branch is pushed

[the remark this answers is not in the document]
Comment (andyjsbell) [reply .github/workflows/persistent-devnet.yml:10]: At the moment I think it makes sense to launch the two workflows on the same trigger. This does deviate from the current release workflow but it does address the need we have to have artefacts being published matching the staging environment that is deployed.

[the remark this answers is not in the document]
Comment (hussein-aitlahcen) [reply .github/workflows/persistent-devnet.yml:10]: I would then say that we push the env on `release-vx.y` and actual release on `release-vx.y.z`.

**Review by vimukthi-git — APPROVED** [review]
LGTM :)

**Review by dzmitry-lahoda — APPROVED** [review]

File: .github/workflows/persistent-devnet.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (vimukthi-git) [reply .github/workflows/persistent-devnet.yml:10]: We would like to not to release on branches without us explicitly doing it sir

[the remark this answers is not in the document]
Comment (andyjsbell) [reply .github/workflows/persistent-devnet.yml:10]: @hussein-aitlahcen @vimukthi-git can we get this resolved please so we can move forward on this one?

[the remark this answers is not in the document]
Comment (hussein-aitlahcen) [reply .github/workflows/persistent-devnet.yml:10]: Yeah I meant that we don't release `vx.y` but `vx.y.z` and use `vx.y` to work on validating the release

**Review by KaiserKarel — APPROVED** [review]

**Comment by hussein-aitlahcen** [comment]
@margifyio refresh

**Comment by hussein-aitlahcen** [comment]
@mergifyio refresh

File: .github/workflows/persistent-devnet.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (vimukthi-git) [reply .github/workflows/persistent-devnet.yml:10]: Depends on what you mean by validate here. If validate=QA , then we always want to have `vx.y.z` tagged release for them. i.e anytime something is in QA means it has been released. If QA certifies that version `vx.y.z` in staging then we go for testnet -> Kusama etc.

**Pull request merged**